src/Worker/Delivery.php

```diff
@@ -199,7 +199,7 @@ public static function execute($cmd, $item_id, $contact_id)
 	 */
 	private static function deliverDFRN($cmd, $contact, $owner, $items, $target_item, $public_message, $top_level, $followup)
 	{
-		logger('Deliver ' . $target_item["guid"] . ' via DFRN to ' . $contact['addr']);
+		logger('Deliver ' . $target_item["guid"] . ' via DFRN to ' . (empty($contact['addr']) ? $contact['url'] : $contact['addr']));
```

Comment:
You can use `defaults()` instead!

Reply:
Hmm, okay.

Reply:
It's not important for this one, just a thought for later.